### ISSUE TITLE
Docker user issue

### ISSUE THREAD

**Issue by hxcadam** [ipsingh06/seedsync#10]
Trying to run in docker it seems to default the user to root as the files coming in are owned by root.  I tried creating the docker with -e PUIG=1001 -e GUID=1001 which is the user that i want to own the files but it was still root that owned them.  I also tried to set it up on ubuntu without docker and never got the pop up to define the user and it defaulted to user 1000.

**Comment by porkie02** [comment]
I Know this is a old post but to fix the root permissions problem is to add a variable umask 000.

**Comment by nitrobass24** [comment]
I am using the PUID & PGID Env Variables without issue on my docker container.

**Comment by flatlinebb** [comment]
@porkie02 , I know this is an old post, but can you explain how you did that in docker?

@nitrobass24 , same question as above. 
Thanks guys!

**Comment by flatlinebb** [comment]
I figured out how to add the PUID and PGID to the Env variables in docker, and they appear to be set without errors, but the downloads are still owned by root:

`PUID=1000`
`PGID=1000`

@nitrobass24 , I would appreciate any additional advice on how to fix this issue, since you seem to have it running correctly.

**Comment by AquaRelliux** [comment]
I also have this issue. Did any of you sort this out?

**Comment by flatlinebb** [comment]
There is no solution at this time that I could find. The downloaded files will be owned by root, so you will have to change the ownership after (or during) the download, with:

`sudo chown user:user files.ext`

Please adjust this for your username and whatever the files are you are downloading.

**Comment by AquaRelliux** [comment]
That is to bad pretty much only thing that is stopping me from using this fully. I will ask some of my docker friends if they know a solution and get back

**Comment by ipsingh06** [comment]
This has been fixed in the latest version v0.7.0. You can specify which user you want to run the docker image as. If no user is specified it will default to user 1000. Please take a look at the README for instructions on specifying the docker user.

**Comment by markcarver** [comment]
I don't think this is actually fixed.

Tried using `--user 1027`, but got a lot of errors in the container saying that user doesn't exist.

**Comment by flatlinebb** [comment]
Try a username, instead of the userid, seems to work for me:
--user bob

**Comment by markcarver** [comment]
No, I need `1027` because that's the actual user id of the `plex` user on my synology NAS. If I just pass `--user plex`, then yes... it creates a "plex" user, but with an id of `1000`; which still doesn't have the proper permissions for the mounted volume. It appears that the owner ids need to match for some reason, cause I can't get `--user` to work at all... have to throw up an "everybody" can read write permission just to get this work at all.

**Comment by flatlinebb** [comment]
Did you try adding the -e switches, with the --user?

-e PUID=1027
-e PGID=1027
--user=plex

Also, try this variant of --user:

--user 1027:1027

The second number is the group number, so make sure it matches what is on
your Synology.

**Comment by markcarver** [comment]
I tried everything you and this thread (and many others) have already suggested... I assure you, I'm no n00b to server config.

AFAICT, the code committed (https://github.com/ipsingh06/seedsync/commit/29e0224cfa1930f0accff061eae7124604c0d9bf#diff-f89d6bc3f2875c3662c33ec518b3911aR26) doesn't actually utilize any environmental variables and manually sets it to `1000` still.

This is the same commit that introduced the verbiage change to mention `--user` in the README.

**Comment by flatlinebb** [comment]
Never implied you were a noob. I apologize if it came across that way.
Might be time to fork the code and fix these small issues. I'm happy to help as time allows.

**Comment by markcarver** [comment]
> Might be time to fork the code and fix these small issues. I'm happy to help as time allows.

Why? It's not like the project is stale... the last commit was 6 days ago lol

**Comment by flatlinebb** [comment]
True. But this topic was closed already. I'm not sure the owner is responsive to our frustrations. I think I'm lucky that my implementation works, because my userid matches the hard-coded one. If I was trying to get it working in a different environment, I'd be pulling my hair out by now. Docker containers are supposed be portable, but this one is proving to be troublesome to say the least.

tl;dr
Fork it because the project owner is not fixing things broken for others.

**Comment by ipsingh06** [comment]
Reopening this since there still seem to be issues.

Folks, docker containers are a pain to work across multiple users. See: https://jtreminio.com/blog/running-docker-containers-as-current-host-user/

@markcarver I'm not sure how using environment variables in the Dockerfile would help. You would have to recompile the image to your specific user anyways. The '--user' parameter is the way Docker supports running as a different user.

I'm unable to reproduce this problem on my end. Posting the error logs from when you use the '--user' option would certainly help. If there's a different way of solving this issue please let me know and I will try it.

**Comment by quadcom** [comment]
Here's the log from Docker on Synology. Doesn't tell me much though.

`

date | stream | content
-- | -- | --
2020-03-02 03:29:02 | stdout | AttributeError: 'NoneType' object has no attribute 'exception'
2020-03-02 03:29:02 | stdout | Seedsync.logger.exception("Caught exception")
2020-03-02 03:29:02 | stdout | File "/app/python/seedsync.py", line 384, in <module>
2020-03-02 03:29:02 | stdout | Traceback (most recent call last):
2020-03-02 03:29:02 | stdout |  
2020-03-02 03:29:02 | stdout | During handling of the above exception, another exception occurred:
2020-03-02 03:29:02 | stdout |  
2020-03-02 03:29:02 | stdout | PermissionError: [Errno 13] Permission denied: '/config/settings.cfg'
2020-03-02 03:29:02 | stdout | with open(file_path, "w") as f:
2020-03-02 03:29:02 | stdout | File "/app/python/common/persist.py", line 53, in to_file
2020-03-02 03:29:02 | stdout | config.to_file(self.config_path)
2020-03-02 03:29:02 | stdout | File "/app/python/seedsync.py", line 59, in __init__
2020-03-02 03:29:02 | stdout | seedsync = Seedsync()
2020-03-02 03:29:02 | stdout | File "/app/python/seedsync.py", line 376, in <module>
2020-03-02 03:29:02 | stdout | Traceback (most recent call last):
2020-03-02 03:24:21 | stdout | AttributeError: 'NoneType' object has no attribute 'exception'
2020-03-02 03:24:21 | stdout | Seedsync.logger.exception("Caught exception")
2020-03-02 03:24:21 | stdout | File "/app/python/seedsync.py", line 384, in <module>
2020-03-02 03:24:21 | stdout | Traceback (most recent call last):
2020-03-02 03:24:21 | stdout |  
2020-03-02 03:24:21 | stdout | During handling of the above exception, another exception occurred:
2020-03-02 03:24:21 | stdout |  
2020-03-02 03:24:21 | stdout | PermissionError: [Errno 13] Permission denied: '/config/settings.cfg'
2020-03-02 03:24:21 | stdout | with open(file_path, "w") as f:
2020-03-02 03:24:21 | stdout | File "/app/python/common/persist.py", line 53, in to_file
2020-03-02 03:24:21 | stdout | config.to_file(self.config_path)
2020-03-02 03:24:21 | stdout | File "/app/python/seedsync.py", line 59, in __init__
2020-03-02 03:24:21 | stdout | seedsync = Seedsync()
2020-03-02 03:24:21 | stdout | File "/app/python/seedsync.py", line 376, in <module>
2020-03-02 03:24:21 | stdout | Traceback (most recent call last):
2020-03-02 03:19:33 | stdout | AttributeError: 'NoneType' object has no attribute 'exception'
2020-03-02 03:19:33 | stdout | Seedsync.logger.exception("Caught exception")
2020-03-02 03:19:33 | stdout | File "/app/python/seedsync.py", line 384, in <module>
2020-03-02 03:19:33 | stdout | Traceback (most recent call last):
2020-03-02 03:19:33 | stdout |  
2020-03-02 03:19:33 | stdout | During handling of the above exception, another exception occurred:
2020-03-02 03:19:33 | stdout |  
2020-03-02 03:19:33 | stdout | PermissionError: [Errno 13] Permission denied: '/config/settings.cfg'
2020-03-02 03:19:33 | stdout | with open(file_path, "w") as f:
2020-03-02 03:19:33 | stdout | File "/app/python/common/persist.py", line 53, in to_file
2020-03-02 03:19:33 | stdout | config.to_file(self.config_path)
2020-03-02 03:19:33 | stdout | File "/app/python/seedsync.py", line 59, in __init__
2020-03-02 03:19:33 | stdout | seedsync = Seedsync()
2020-03-02 03:19:33 | stdout | File "/app/python/seedsync.py", line 376, in <module>
2020-03-02 03:19:33 | stdout | Traceback (most recent call last):
2020-03-02 03:19:00 | stdout | AttributeError: 'NoneType' object has no attribute 'exception'
2020-03-02 03:19:00 | stdout | Seedsync.logger.exception("Caught exception")
2020-03-02 03:19:00 | stdout | File "/app/python/seedsync.py", line 384, in <module>
2020-03-02 03:19:00 | stdout | Traceback (most recent call last):
2020-03-02 03:19:00 | stdout |  
2020-03-02 03:19:00 | stdout | During handling of the above exception, another exception occurred:
2020-03-02 03:19:00 | stdout |  
2020-03-02 03:19:00 | stdout | PermissionError: [Errno 13] Permission denied: '/config/settings.cfg'
2020-03-02 03:19:00 | stdout | with open(file_path, "w") as f:
2020-03-02 03:19:00 | stdout | File "/app/python/common/persist.py", line 53, in to_file
2020-03-02 03:19:00 | stdout | config.to_file(self.config_path)
2020-03-02 03:19:00 | stdout | File "/app/python/seedsync.py", line 59, in __init__
2020-03-02 03:19:00 | stdout | seedsync = Seedsync()
2020-03-02 03:19:00 | stdout | File "/app/python/seedsync.py", line 376, in <module>
2020-03-02 03:19:00 | stdout | Traceback (most recent call last):
2020-03-02 03:01:22 | stdout | AttributeError: 'NoneType' object has no attribute 'exception'
2020-03-02 03:01:22 | stdout | Seedsync.logger.exception("Caught exception")
2020-03-02 03:01:22 | stdout | File "/app/python/seedsync.py", line 384, in <module>
2020-03-02 03:01:22 | stdout | Traceback (most recent call last):
2020-03-02 03:01:22 | stdout |  
2020-03-02 03:01:22 | stdout | During handling of the above exception, another exception occurred:
2020-03-02 03:01:22 | stdout |  
2020-03-02 03:01:22 | stdout | PermissionError: [Errno 13] Permission denied: '/config/settings.cfg'
2020-03-02 03:01:22 | stdout | with open(file_path, "w") as f:
2020-03-02 03:01:22 | stdout | File "/app/python/common/persist.py", line 53, in to_file
2020-03-02 03:01:22 | stdout | config.to_file(self.config_path)
2020-03-02 03:01:22 | stdout | File "/app/python/seedsync.py", line 59, in __init__
2020-03-02 03:01:22 | stdout | seedsync = Seedsync()
2020-03-02 03:01:22 | stdout | File "/app/python/seedsync.py", line 376, in <module>
2020-03-02 03:01:22 | stdout | Traceback (most recent call last):
2020-03-02 02:55:03 | stdout | AttributeError: 'NoneType' object has no attribute 'exception'
2020-03-02 02:55:03 | stdout | Seedsync.logger.exception("Caught exception")
2020-03-02 02:55:03 | stdout | File "/app/python/seedsync.py", line 384, in <module>
2020-03-02 02:55:03 | stdout | Traceback (most recent call last):
2020-03-02 02:55:03 | stdout |  
2020-03-02 02:55:03 | stdout | During handling of the above exception, another exception occurred:
2020-03-02 02:55:03 | stdout |  
2020-03-02 02:55:03 | stdout | PermissionError: [Errno 13] Permission denied: '/config/settings.cfg'
2020-03-02 02:55:03 | stdout | with open(file_path, "w") as f:
2020-03-02 02:55:03 | stdout | File "/app/python/common/persist.py", line 53, in to_file
2020-03-02 02:55:03 | stdout | config.to_file(self.config_path)
2020-03-02 02:55:03 | stdout | File "/app/python/seedsync.py", line 59, in __init__
2020-03-02 02:55:03 | stdout | seedsync = Seedsync()
2020-03-02 02:55:02 | stdout | File "/app/python/seedsync.py", line 376, in <module>
2020-03-02 02:55:02 | stdout | Traceback (most recent call last):

`

**Comment by ipsingh06** [comment]
Please try the newest version (v0.7.1). I've added support for running docker images with any host uid.

**Comment by sinwe** [comment]
@ipsingh06 how to configure different uid and gid in docker-compose.yml? I tried with environment PUID and UID as well as user, all doesn't work

**Comment by quadcom** [comment]
Only way I've gotten it to work in Unraid is to pass it in as an extra parameter in advanced mode.

--user="99:100"

Still doesn't solve the umask problem though.

**Comment by sinwe** [comment]
> Only way I've gotten it to work in Unraid is to pass it in as an extra parameter in advanced mode.
> 
> --user="99:100"
> 
> Still doesn't solve the umask problem though.

You mean when running docker-compose command I have to pass in `--user=uid:gid`? That will affect every container in my `docker-compose.yml`
